x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/types.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CriteriaWithPagination } from '@elastic/eui';
+
+export interface Meta {
```

Comment:
suggestion: You'll want to use our Paginate type instead of the Meta type, that's a little outdated. The Paginate type should be easier to work with, and can just be plugged into a table directly with very little modification.

https://github.com/elastic/kibana/tree/main/x-pack/plugins/enterprise_search/common/types/pagination.ts

use example: https://github.com/elastic/kibana/tree/main/x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/sync_jobs/sync_jobs.tsx

Reply:
Little bit confused on implementing Paginate type and have few questions as well. 

My understanding so far: 

Current API returns 
```
meta: {
     from: number;
     size: number;
     total: number;
   };
results: {
   name: string;
   indices: string[];
}
```
Paginate type variable would return 

```
{
  data: [],
  has_more_hits_than_total: false,
  pageIndex: 0,
  pageSize: 10,
  size: 0,
  total: 0,
}
```

From my understanding `data` field in the paginate variable return the actual fetchEngine results and we would get the pagination data from this paginate variable. which basically means the variable of type Paginate would be the one that makes the request to backend. 

1. My assumptions is that, to implement Paginate type requires more change than a change in  Engine List Table ?

2. Do we need to wait for Kibana API call to be ready ?

Reply:
Ah, I had missed that engines are still going to live in ent-search and be called with ent-search APIs, rather than Kibana APIs that call Elasticsearch directly. My bad, ignore my ramblings here!